src/web/waiters/RecipeWaiter.mjs

```diff
@@ -580,29 +590,70 @@ class RecipeWaiter {
 
 
     /**
-     * Adjusts the number of ingredient columns as the width of the recipe changes.
+     * Remove all "selected" classes for op-list list items at once
+     *
+     * This hides all the checkmark icons of previously added ( selected )
+     * operations to the recipe list
      */
-    adjustWidth() {
-        const recList = document.getElementById("rec-list");
-
-        // Hide Chef icon on Bake button if the page is compressed
-        const bakeIcon = document.querySelector("#bake img");
-
-        if (recList.clientWidth < 370) {
-            // Hide Chef icon on Bake button
-            bakeIcon.style.display = "none";
-        } else {
-            bakeIcon.style.display = "inline-block";
+    clearAllSelectedClasses() {
+        const list = document.querySelectorAll(".operation.selected");
+
+        // check if any operations are selected at all to prevent errors
+        if (list.length) {
+            list.forEach((item) => {
+                item.classList.remove("selected");
+            });
         }
+    }
 
-        // Scale controls to fit pane width
-        const controls = document.getElementById("controls");
-        const controlsContent = document.getElementById("controls-content");
-        const scale = (controls.clientWidth - 1) / controlsContent.scrollWidth;
-
-        controlsContent.style.transform = `scale(${scale})`;
+    /**
+     * Add "selected" to the operation that is added to the recipe
+     *
+     * This displays a checkmark icon to the operation in the op-list
+     *
+     * @param {string} opDataName the data-name of the target operation
+     */
+    addSelectedClass(opDataName) {
+        const list = document.querySelectorAll(".operation");
+        const item = Array.from(list).filter((item) => item.getAttribute("data-name") === opDataName);
+
+        // when an item is listed in favourites, there are 2 of
+        // them and both need the 'selected' class ( checkmark )
+        item.forEach((op) => {
+            op.classList.add("selected");
+        });
     }
 
+    /**
+     * Update which items are selected in op-list.
+     *
+     * First, all selected classes are removed from op-list, then we get the current
+     * recipe-list ingredient names and add 'selected' back to the matching operations.
+     *
+     * Note: It seems a little overkill, but with the current tightly coupled code this is
+     * a reliable way to make sure the 'selected' operations are always in sync with
+     * the recipe list ( I think this is preferable to complicating a lot of existing
+     * code ), I'd recommend to refactor this at one point, but that would mean a huge code
+     * overhaul for another time / issue.
+     */
+    updateSelectedOperations() {
+        const recipeList = document.querySelectorAll("#rec-list > li");
```

Comment:
I know it's nitpicky, but you're selecting list-items here and not the list itself. Rename the variable to `recipes` or `recipeListItems`